gst/nnstreamer/nnstreamer_plugin_api_filter.h

```diff
@@ -184,41 +252,81 @@ typedef struct _GstTensorFilterFramework
-       * @return 0 if ok. < 0 if error.
-       */
-
-  void (*close) (const GstTensorFilterProperties * prop, void **private_data);
-      /**< Optional. tensor_filter.c will not call other callbacks after calling close. Free-ing private_data is this function's responsibility. Set NULL after that.
-       *
-       * @param[in] prop read-only property values
-       * @param[in/out] private_data A subplugin may save its internal private data here. The subplugin is responsible for alloc/free of this pointer. Normally, close() frees private_data and set NULL.
-       */
-
-  void (*destroyNotify) (void * data);
+      void (*destroyNotify) (void * data);
       /**< Optional. tensor_filter.c will call it when 'allocate_in_invoke' flag of the framework is TRUE. Basically, it is called when the data element is destroyed. If it's set as NULL, g_free() will be used as a default. It will be helpful when the data pointer is included as an object of a nnfw. For instance, if the data pointer is removed when the object is gone, it occurs error. In this case, the objects should be maintained for a while first and destroyed when the data pointer is destroyed. Those kinds of logic could be defined at this method.
        *
        * @param[in] data the data element.
        */
 
-  int (*reloadModel) (const GstTensorFilterProperties * prop, void **private_data);
+      int (*reloadModel) (const GstTensorFilterProperties * prop, void **private_data);
       /**< Optional. tensor_filter.c will call it when a model property is newly configured. Also, 'is-updatable' property of the framework should be TRUE. This function reloads a new model specified in the 'prop' argument. Note that it requires extra memory size enough to temporarily hold both old and new models during this function to hide the reload overhead.
        *
        * @param[in] prop read-only property values
        * @param[in/out] private_data A subplugin may save its internal private data here. The subplugin is responsible for alloc/free of this pointer. Normally, close() frees private_data and set NULL.
        * @return 0 if ok. < 0 if error.
        */
 
-  int (*checkAvailability) (accl_hw hw);
+      int (*checkAvailability) (accl_hw hw);
       /**< Optional. Check if the provided hardware accelerator is supported. This check is static or dynamic based on framework support. Positive response of this check does not guarantee successful running of model with this accelerator. The static check can be performed without opening the framework.
        *
        * @param[in] hw backend accelerator hardware
        * @return 0 if supported. -errno if not supported.
        */
+    };
+
+    /**
+     * @brief Tensor_Filter Subplugin definition Version 2
+     */
+    struct /** _GstTensorFilterFramework_v2 */
+    {
+      int (*getFrameworkInfo) (void **private_data, GstTensorFilterFrameworkInfo *fw_info);
```

Comment:
I don't think private_data is available for getFrameworkInfo.
prop will be available at that time, though.

Reply:
private_data is optional (it can be NULL).
this is needed for custom filter subplugin as `allocate_in_invoke` property for custom filter is finalized based on the model. 
OR 
remove private_data from here, and move `allocate_in_invoke` property to getModelInfo (note that this property is model dependent only for custom filter subplugin).

Reply:
Ah.. ok.. then, let's keep both prop and private_data. and write a "caution" statement as a comment that it could be NULL if it's not yet opened.

Reply:
Added an extra caution statement apart from argument description.